src/sentry/api/endpoints/organization_events_meta.py

```diff
@@ -172,17 +173,91 @@ class OrganizationSpansSamplesEndpoint(OrganizationEventsEndpointBase):
 
     def get(self, request: Request, organization) -> Response:
         is_frontend = is_frontend_request(request)
+        sample_count = 9
+
         try:
             snuba_params = self.get_snuba_params(request, organization)
         except NoProjects:
             return Response({})
 
+        use_rpc = request.GET.get("useRpc", "0") == "1"
+
         buckets = request.GET.get("intervals", 3)
         lower_bound = request.GET.get("lowerBound", 0)
         first_bound = request.GET.get("firstBound")
         second_bound = request.GET.get("secondBound")
         upper_bound = request.GET.get("upperBound")
         column = request.GET.get("column", "span.self_time")
+
+        if use_rpc:
```

Comment:
Mind moving the block below this to a function

Reply:
done!